AlphaWallet/Settings/ViewModels/ExportJsonKeystorePasswordViewModel.swift

```diff
@@ -0,0 +1,56 @@
+//
+//  ExportJsonKeystorePasswordViewModel.swift
+//  AlphaWallet
+//
+//  Created by Jerome Chan on 2/12/21.
+//
+
+import Foundation
+
+typealias ExportJsonKeystorePasswordViewModelCompletion = (Result<String, KeystoreError>) -> Void
+
+class ExportJsonKeystorePasswordViewModel {
+    private let validator: StringValidator
+    private let keystore: Keystore
+
+    init(keystore: Keystore) {
+        self.validator = StringValidator(rules: [
+            .lengthMoreThanOrEqualTo(6),
+            .canOnlyContain(CharacterSet.alphanumerics)
+        ])
+        self.keystore = keystore
+    }
+
+    init(keystore: Keystore, validator: StringValidator) {
+        self.validator = validator
+        self.keystore = keystore
+    }
+
+    func validate(password: String) -> StringValidatorResult {
+        return validator.validate(string: password)
+    }
+
+    func containsIllegalCharacters(password: String) -> Bool {
+        return validator.containsIllegalCharacters(string: password)
+    }
+
+    func computeHash(password: String, completion: @escaping ExportJsonKeystorePasswordViewModelCompletion) {
```

Comment:
Do you want to try returning a Promise here instead? We still have lots of completion blocks all over the place, but generally for those that only evaluate once, we should try to use `PromiseKit`. PromiseKit wrapped code is a little more troublesome to write than completion-based ones, but the client code is much clearer.